### PR TITLE
Developer targets

### PR DESCRIPTION
**Pull request by Wenzel**
Solve https://github.com/IntelLabs/kAFL/issues/79

### CODE REVIEW

**Comment by il-steffen** [comment]
Still need to remove force flags on nyx_packer and qemu tasks: https://github.com/IntelLabs/kAFL/search?q=force

File: deploy/Makefile [hunk 1]
```diff
@@ -40,3 +41,15 @@ venv:
...
 clean:
 	rm -rf "$(ROOT_DIR)/venv"
+
+# Developer targets
+#------------------
+# pull the latest changes from all components
+# assumes local deployment
+update:
+	venv/bin/ansible-galaxy install -r requirements.yml --force
+	$(ANSIBLE_COMMAND) -K --tags clone
```
Comment (il-steffen) [inline deploy/Makefile:51]: add $(EXTRA_ARGS) so that we can do `make update -- --list-tasks`

File: deploy/Makefile [hunk 2]
```diff
@@ -40,3 +41,15 @@ venv:
...
+
+# rebuild all components
+build:
+	$(ANSIBLE_COMMAND) -K --tags build
```
Comment (il-steffen) [inline deploy/Makefile:55]: add $(EXTRA_ARGS) so that we can do make build -- --list-tasks

**Review by il-steffen — CHANGES_REQUESTED** [review]

**Comment by il-steffen** [comment]
- both, `update` and `build` should depend on venv and I think we should move `ansible-galaxy install` also into venv or separate rule to test+install if needed (which in turn depends on venv)
- it looks like we're mostly duplicating logic in deploy/Makefile so maybe we just merge them

**Comment by Wenzel** [comment]
> I think we should move ansible-galaxy install also into venv

This will cause issue when updating the intellabs.kafl collection locally for testing. Since the venv is already created, there will be no update of the intellabs.kafl collection installation when running make `deploy`.

The only solution I found is to include it before running each target.

> it looks like we're mostly duplicating logic in deploy/Makefile so maybe we just merge them

that's a possibility for a future PR

**Comment by il-steffen** [comment]
still need to remove remaining `force` flags? https://github.com/IntelLabs/kAFL/search?q=force

**Comment by Wenzel** [comment]
> still need to remove remaining `force` flags? https://github.com/IntelLabs/kAFL/search?q=force

Remving the force clone flag for QEMU will lead to a fail redeploy, since QEMU build leads the repo into a dirty state.
Either we leave the flag for QEMU, or add the fix that I mentioned here:
https://github.com/IntelLabs/kAFL/issues/79#issuecomment-1170391485

File: deploy/Makefile [hunk 3]
```diff
@@ -31,6 +31,7 @@ ifeq ($(LOCALHOST_INVENTORY),0)
 	@echo "┏━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓"
 	@echo "┃ Ansible BECOME password: if you are using a passwordless SUDO, skip by pressing enter.┃"
 	@echo "└━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┘"
+	@echo "Note: Git repositories will be resetted. Uncommitted changes will be lost."
```
Comment (il-steffen) [inline deploy/Makefile:34]: "Git repositories will be reset."

**Review by il-steffen — APPROVED** [review]

**Pull request merged**